rest/historical_test_data_routes.go

```diff
@@ -0,0 +1,348 @@
+	if err != nil {
+		return err
+	}
+
+	return err
+}
+
+// readRequesters parses requesters parameter values and translates them into a
+// list of internal Evergreen requester names.
+func (h *htdFilterHandler) readRequesters(requesters []string) ([]string, error) {
+	if len(requesters) == 0 {
+		requesters = []string{htdAPIRequesterMainline}
+	}
+
+	requesterValues := []string{}
+	for _, requester := range requesters {
+		switch requester {
+		case htdAPIRequesterMainline:
+			requesterValues = append(requesterValues, cedar.RepotrackerVersionRequester)
+		case htdAPIRequesterPatch:
+			requesterValues = append(requesterValues, cedar.PatchRequesters...)
+		case htdAPIRequesterTrigger:
+			requesterValues = append(requesterValues, cedar.TriggerRequester)
+		case htdAPIRequesterGitTag:
+			requesterValues = append(requesterValues, cedar.GitTagRequester)
+		case htdAPIRequesterAdhoc:
+			requesterValues = append(requesterValues, cedar.AdHocRequester)
+		default:
+			return nil, errors.Errorf("invalid requester value %v", requester)
+		}
+	}
+	return requesterValues, nil
+}
+
+// readStringList parses a string list parameter value, the values can be comma
+// separated or specified multiple times.
+func (h *htdFilterHandler) readStringList(values []string) []string {
+	var parsedValues []string
+	for _, val := range values {
+		elements := strings.Split(val, ",")
+		parsedValues = append(parsedValues, elements...)
+	}
+	return parsedValues
+}
+
+// readInt parses an integer parameter value, given minimum, maximum, and
+// default values.
+func (h *htdFilterHandler) readInt(intString string, min, max, defaultValue int) (int, error) {
+	if intString == "" {
+		return defaultValue, nil
+	}
+
+	value, err := strconv.Atoi(intString)
+	if err != nil {
+		return 0, err
+	}
+
+	if value < min || value > max {
+		return 0, errors.New("invalid int parameter value")
+	}
+	return value, nil
+}
+
+// readTestGroupBy parses a sort parameter value and returns the corresponding
+// HTDSort struct.
+func (h *htdFilterHandler) readSort(sortValue string) (model.HTDSort, error) {
+	switch sortValue {
+	case htdAPISortEarliest:
+		return model.HTDSortEarliestFirst, nil
+	case htdAPISortLatest:
+		return model.HTDSortLatestFirst, nil
+	case "":
+		return model.HTDSortEarliestFirst, nil
+	default:
+		return model.HTDSort(""), gimlet.ErrorResponse{
+			Message:    "invalid sort value",
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+}
+
+// readGroupBy parses a group_by parameter value and returns the corresponding
+// HTDGroupBy struct.
+func (h *htdFilterHandler) readGroupBy(groupByValue string) (model.HTDGroupBy, error) {
+	switch groupByValue {
+	case htdAPITestGroupByVariant:
+		return model.HTDGroupByVariant, nil
+	case htdAPITestGroupByTask:
+		return model.HTDGroupByTask, nil
+	case htdAPITestGroupByTest:
+		return model.HTDGroupByTest, nil
+	// We no longer store distro, but do not want to error if it is
+	// present.
+	case "", htdAPITestGroupByDistro:
```

Comment:
Is this a backwards compatibility case? Is it not possible to do this distro->variant translation in the Evergreen REST route rather than preserve its clunky behavior here?

Reply:
Yeah it is for backwards compatibility, we can still do that translation in the Evergreen REST route but I think it's fine to keep it just in case.